drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableGraphEditorPresenter.java

```diff
@@ -396,6 +406,81 @@ public void makeMenuBar() {
                 .build();
     }
 
+    protected Command getSaveAndRenameCommand() {
+
+        return saveAndRenameCommandBuilder
+                .addPathSupplier(getPathSupplier())
+                .addValidator(getValidator())
+                .addRenameService(getGraphSaveAndRenameService())
+                .addMetadataSupplier(getMetadataSupplier())
+                .addContentSupplier(getContentSupplier())
+                .addIsDirtySupplier(getIsDirtySupplier())
+                .addSuccessCallback(onSuccess())
+                .build();
+    }
+
+    Supplier<Path> getPathSupplier() {
+        return () -> versionRecordManager.getPathToLatest();
+    }
+
+    AssetUpdateValidator getValidator() {
+        return assetUpdateValidator;
+    }
+
+    Caller<GuidedDecisionTableGraphSaveAndRenameService> getGraphSaveAndRenameService() {
+        return graphSaveAndRenameService;
+    }
+
+    Supplier<Metadata> getMetadataSupplier() {
+        return () -> {
+
+            final Overview overview = getActiveDocument().getOverview();
+
+            return overview.getMetadata();
+        };
+    }
+
+    Supplier<List<GuidedDecisionTableEditorContent>> getContentSupplier() {
+        return () -> getAvailableDecisionTables()
+                .stream()
+                .map(c -> new GuidedDecisionTableEditorContent(c.getModel(), c.getOverview(), c.getCurrentPath(), c.getLatestPath()))
+                .collect(Collectors.toList());
+    }
+
+    Supplier<Boolean> getIsDirtySupplier() {
+
+        return () -> getAvailableDecisionTables().stream().anyMatch(dtPresenter -> {
```

Comment:
See my comments about the "single" GDT editor above. This makes perfect sense here.

However, in addition to a GDT being "dirty" the "graph" can also be dirty (as it has it's own metadata). You probably also need to consider ```originalGraphHash``` too?

Reply:
Good catch! Thank you :-) Fixed.